shapely/coordinates.py

```diff
@@ -20,11 +28,26 @@ def transform(geometry, transformation, include_z=False):
     transformation : function
         A function that transforms a (N, 2) or (N, 3) ndarray of float64 to
         another (N, 2) or (N, 3) ndarray of float64.
-    include_z : bool, default False
-        If True, include the third dimension in the coordinates array
-        that is passed to the ``transformation`` function. If a
-        geometry has no third dimension, the z-coordinates passed to the
-        function will be NaN.
+        The function may not change N.
+    include_z : bool, optional, default False
+        If False, always return 2D geometries.
+        If True, the data being passed to the
+        transformation function will include the third dimension
+        (if a geometry has no third dimension, the z-coordinates
+        will be NaN). If None, will infer the dimensionality per
+        input geometry using ``has_z``, which may result in 2 calls to
+        the transformation function. Note that this inference
+        can be unreliable with empty geometries or NaN coordinates: for a
+        guaranteed result, it is recommended to specify ``include_z`` explicitly.
+    interleaved : bool, default True
+        If set to False, the transformation function should accept 2 or 3 separate
+        one-dimensional arrays (x, y and optional z) instead of a single
+        two-dimensional array.
+
+    See Also
+    --------
+    has_z : Returns a copy of a geometry array with a function applied to its
+        coordinates.
```

Comment:
This is a wrong copy-paste I assume, it's the explanation of `transform` itself, not `has_z`